pynxtools/dataconverter/readers/transmission/reader.py

```diff
@@ -149,25 +146,36 @@ def read_detectors(metadata: list) -> Dict[str, Any]:
     detector_times = parse_detector_line(metadata[32], float)
     detector_gains = parse_detector_line(metadata[35], float)
     detector_changes = [float(x) for x in metadata[43].split()]
-    wavelength_ranges = \
+    wavelength_ranges = (
         [mpars.MIN_WAVELENGTH] + detector_changes[::-1] + [mpars.MAX_WAVELENGTH]
+    )
 
     template.update(
         convert_detector_to_template(
-            "PMT", detector_slits[2], detector_times[2],
-            None, 2, [wavelength_ranges[0], wavelength_ranges[1]]
+            "PMT",
+            detector_slits[-1],
+            detector_times[-1],
+            None,
+            2,
+            [wavelength_ranges[0], wavelength_ranges[1]],
         )
     )
 
-    for name, idx in zip(["PbS", "InGaAs"], [1, 0]):
+    for name, idx, slit, time, gain in zip(
+        ["PbS", "InGaAs"],
+        [1, 0],
+        detector_slits[1:],
+        detector_times[1:],
+        detector_gains[1:],
```

Comment:
I'm guessing this is the important change here? This looks fine how its being used below. If so, this is ready to merge. I'll approve this unless something else needs attention to.

Reply:
Yeah, this and the other line I added a comment for. Sorry that I committed my auto-formatting here, too